app/mailers/user_mailer.rb

```diff
@@ -16,25 +16,38 @@ def reservation_status_update(reservation, receipt = false) # rubocop:disable al
     @reservation = reservation
     @status = @reservation.status
 
-    return if !receipt && @status == 'returned overdue' &&
+    return if !receipt && @status == 'returned' && @reservation.overdue &&
               @reservation.equipment_model.late_fee == 0
 
     return if receipt && @reservation.checked_out.nil?
 
+    if @reservation.overdue
+      if @status == 'returned'
+        @status = 'returned overdue'
+      else
+        @status = 'overdue'
+      end
+    end
+
```

Comment:
Yea, that's fair (I wasn't really disagreeing with you). I guess the problem of determining an e-mail status is basically the same as generating a unique status string based on all the different status possibilities, and making that a model method instead of some code in the Rake task makes sense as well.